### PR TITLE
Migrate password change endpoint

### LINKED ISSUE #18: Migrated Endpoint to change password
- Corresponding API tag: v0.4.X
- Migrated password change endpoint from UserService to AuthenticationService
- Endpoint is renamed to: /authentication-management/users/{username}
- change password is now a put instead of a post

- Lagom:
  - [ ] implemented functional behaviour [Lagom tag v0.4.X]
  - [ ] implemented completely [Lagom tag v0.4.X]
  - [ ] unit tested [Lagom tag v0.4.X]

- Vue: 
  - [ ] implemented functional behaviour [Vue tag vX.X.X]
  - [ ] implemented completely [Vue tag vX.X.X]
  - [ ] unit tested [Vue tag vX.X.X]

### PR DESCRIPTION
**Pull request by ldklenner**
### Reason for this PR
- The AuthenticationService should be offer the password change functionality

### Changes in this PR
- Migrate password change endpoint from UserService to AuthenticationService

### CODE REVIEW

File: authentication-management.yaml [hunk 1]
```diff
@@ -26,13 +32,130 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/JsonVersionNumber'
+  /users/{username}:
+    post:
```
Comment (david-buderus) [inline authentication-management.yaml]: This needs to be a PUT

**Review by david-buderus — CHANGES_REQUESTED** [review]

**Review by david-buderus — APPROVED** [review]

**Review by bastihav — APPROVED** [review]
We now have two definitions of AuthUser (user service needs it to create users), we should import the authentication management authUser into the user management API.

**Comment by ldklenner** [comment]
Yes, that would be convenient, but I am not sure how to do this. When editing the yamls I looked at the existing API, like the `DetailedError` is defined in multiple API specification like `course.yaml` or `user-management.yaml` I copied the `AuthUser`.

File: authentication-management.yaml [hunk 2]
```diff
@@ -26,13 +32,130 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/JsonVersionNumber'
+  /users/{username}:
```
Comment (bastihav) [inline authentication-management.yaml:35]: is this really a descriptive route naming btw?

Comment (ldklenner) [reply authentication-management.yaml:35]: No, the name is likely to change in the future

**Pull request merged**